lib/internal/webstreams/readablestream.js

```diff
@@ -653,7 +653,7 @@ function TransferredReadableStream() {
   const stream = new InternalTransferredReadableStream();
 
   stream.constructor = ReadableStream;
-
+  ObjectSetPrototypeOf(stream, ReadableStream.prototype);
```

Comment:
We should probably do the same inside [`createReadableStream`](https://github.com/nodejs/node/blob/11dce6c69c00f582b8754ca318ea5f4c517221a5/lib/internal/webstreams/readablestream.js#L1247-L1253) and [`createReadableByteStream`](https://github.com/nodejs/node/blob/11dce6c69c00f582b8754ca318ea5f4c517221a5/lib/internal/webstreams/readablestream.js#L1274-L1280). Same for [`createWritableStream` in `writablestream.js`](https://github.com/nodejs/node/blob/11dce6c69c00f582b8754ca318ea5f4c517221a5/lib/internal/webstreams/writablestream.js#L540-L546)

Or maybe we should move the `ObjectSetPrototypeOf` calls inside these `Internal*Stream` constructors? Would it be better for performance to change the prototype as early as possible?

Reply:
I did a rough benchmark, and it didn't show much perf improvement, but it's good suggestion - putting it in constructor is more proper.

Reply:
by "rough", I mean 😅
```
console.time('1')
for (let i = 0; i < 10_000; i++) {
  const r1 = new ReadableStream(); 
  structuredClone(r1, { transfer: [r1] });
  
  const w1 = new WritableStream();
  structuredClone(w1, { transfer: [w1] });
  
  const t1 = new TransformStream();
  structuredClone(t1, { transfer: [t1] });
}
console.timeEnd('1')
```

Comparing this with the suggested changes, I don't see much difference. A benchmark CI summary would be more accurate I think.